parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,216 @@
+            implements FileVisitor<Path>
+    {
+        private void processStatement(BufferedReader reader)
+                throws IOException
+        {
+            StringBuilder statement = new StringBuilder();
+            while (true) {
+                final String line = reader.readLine();
+                if (line == null || isLineAfterStatement(line)) {
+                    break;
+                }
+                statement.append(line);
+            }
+            if (statement.length() > 0 && !statement.substring(statement.length() - 1).equals(";")) {
+                statement.append(";"); // NOTE: This semi-colon is necessary for parsing.
+            }
+            result.incrementNumStatementsFound();
+            validateParsingOfSqlStatement(statement.toString());
+        }
+
+        private boolean isLineBeforeStatement(String line)
+        {
+            return line.startsWith("statement ok") || line.startsWith("query");
+        }
+
+        private boolean statementIsAcceptableToCurrRdms()
+        {
+            if (config.getRdbms() == ALL) {
+                return skipIfSet.isEmpty() && onlyIf == null;
+            }
+
+            if (onlyIf != null && onlyIf != config.getRdbms()) {
+                return false;
+            }
+            return !skipIfSet.contains(config.getRdbms());
+        }
+
+        private boolean isLineAfterStatement(String line)
+        {
+            return line.equals("----") || line.equals("");
+        }
+
+        private boolean processConditionalRecord(String line)
+        {
+            String[] splitLine = line.split(" ");
+            if (splitLine.length <= 1) {
+                return false;
+            }
+
+            if (line.startsWith("onlyif")) {
+                String onlyIfRdms = splitLine[1];
+                for (AcceptedRdbms acceptedRdbms : AcceptedRdbms.values()) {
+                    if (acceptedRdbms.toString().equalsIgnoreCase(onlyIfRdms)) {
+                        onlyIf = acceptedRdbms;
+                    }
+                }
+                return true;
+            }
+
+            if (line.startsWith("skipif")) {
+                String skipIfRdms = splitLine[1];
+                for (AcceptedRdbms acceptedRdbms : AcceptedRdbms.values()) {
+                    if (acceptedRdbms.toString().equalsIgnoreCase(skipIfRdms)) {
+                        skipIfSet.add(acceptedRdbms);
+                    }
+                }
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+        {
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException
+        {
+            if (Files.isHidden(file) || result.getNumFilesVisited() > config.getMaxNumFilesToVisit()) {
```

Comment:
If we are greater than max num files to visit, aren't we supposed to terminate?

Reply:
Yes.

Changed to terminate if we're greater than max num files to visit.

Done.